insights/parsers/initscript.py

```diff
@@ -0,0 +1,104 @@
+"""
+InitScript - files ``/etc/rc.d/init.d/*``
+=========================================
+
+InitScript is a parser for the initscripts in ``/etc/rc.d/init.d``.
+
+Because this parser read multiple files, the initscripts are stored as a list
+within the parser and need to be iterated through in order to find specific
+initscripts.
+
+Examples:
+
+    >>> for initscript in shared[InitScript]: # Parser contains list of all initscripts
+    ...     print "Name:", initscript.file_name
+    ...
+    Name: netconsole
+    Name: rhnsd
+
+"""
+
+from . import SkipException
+from .. import Parser, parser
+import re
+
+CHKCONFIG_REGEX = re.compile(r"^#\s+chkconfig:\s+.*$")
+LSB_REGEX = re.compile(r"^#\s+Provides:\s+.*$")
+
+SHEBANG_REGEX = re.compile(r"^#!\s*/.*$")
+STARTSTOPSTATUS_REGEX = re.compile(r"\b(start|stop|status)\b")
+
+
+class EmptyFileException(SkipException):
+    pass
+
+
+class NotInitscriptException(SkipException):
```

Comment:
@rmetrich I apologize for not noticing this in the first pass, but you should be using `ParseException` instead of `SkipException` here.  See this document for the difference http://insights-core.readthedocs.io/en/1.x/exception_model.html#parsing-and-exceptions